src/providers/wms/qgswmsprovider.cpp

```diff
@@ -3018,8 +3018,22 @@ QgsRasterIdentifyResult QgsWmsProvider::identify( const QgsPointXY &point, QgsRa
     if ( mTileLayer->getFeatureInfoURLs.contains( formatStr ) )
     {
       // REST
+
+
       QString url = mTileLayer->getFeatureInfoURLs[ formatStr ];
 
+      if ( mSettings.mIgnoreGetFeatureInfoUrl )
+      {
+        // rewrite the URL if the one in the capabilities document is incorrect
+        // strip every thing after the ? from the base url
+        const QString base = mSettings.mBaseUrl.split( QRegularExpression( "\\?" ) )[0];
```

Comment:
Sorry I missed this before: accessing a string list without checking its size is dangerous (it can segfault). Can you check the size of the result of split before accessing the first element for example you can use https://doc.qt.io/qt-5/qlist.html#isEmpty ? For clarity you can also use https://doc.qt.io/qt-5/qlist.html#first to get the first element.